userd/autostart.go

```diff
@@ -35,10 +34,25 @@ import (
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/logger"
 	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
 	"github.com/snapcore/snapd/strutil/shlex"
 	"github.com/snapcore/snapd/systemd"
 )
 
+var (
+	currentDesktop = splitSkippingEmpty(os.Getenv("XDG_CURRENT_DESKTOP"), ":")
+)
+
+func splitSkippingEmpty(s, substr string) []string {
```

Comment:
We could use `filepath.SplitList()` here but that won't skip empty so maybe not.